src/main/java/org/springframework/integration/kafka/dsl/KafkaMessageDrivenChannelAdapterSpec.java

```diff
@@ -141,6 +142,18 @@ public S recoveryCallback(RecoveryCallback<? extends Object> recoveryCallback) {
 		return _this();
 	}
 
+	/**
+	 * When using a type-aware message converter (such as {@code StringJsonMessageConverter},
+	 * set the payload type the converter should create. Defaults to {@link Object}.
+	 * @param payloadType the type.
+	 * @return the spec
+	 * @since 3.2.0
+	 */
+	public S payloadType(Class payloadType) {
+		this.target.setPayloadType(payloadType);
```

Comment:
Thanks. 

any chance to have a simple test-case to cover this change?

Reply:
I split another json test that would set the payload type after the dsl. The new test sets the payload type in the dsl and has the same assertions the previous test did in its second half. Let me know if this is what you were looking for.